README.md
# BigDataCA4

changes.txt
------------------------------------------------------------------------
r1551925 | Thomas | 2015-11-27 16:57:44 +0000 (Fri, 27 Nov 2015) | 1 line
Changed paths:
   A /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client/res/drawable-xxxhdpi (from /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client/res/drawablw-xxxhdpi:1551688)
   D /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client/res/drawablw-xxxhdpi
   A /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-bt/res/drawable-xxxhdpi (from /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-bt/res/drawablw-xxxhdpi:1551922)
   D /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-bt/res/drawablw-xxxhdpi

Renamed folder to the correct name
------------------------------------------------------------------------
r1551575 | Thomas | 2015-11-27 09:46:32 +0000 (Fri, 27 Nov 2015) | 1 line
Changed paths:
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/common/mvn-config/res/values/environment.xml
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/mct/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/mct/build-config/prod_lowbandwidth.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/mct/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/mct/build-config/qa_lowbandwidth.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/mct/build-config/voucher.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client/build-config/dev.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-amx/build-config/dev.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-amx/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-amx/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-bt/build-config/dev.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-bt/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-bt/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-frt/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-frt/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-sfr/build-config/dev.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-sfr/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-sfr/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-spanishclaros/build-config/dev.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-spanishclaros/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-spanishclaros/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-telcel/build-config/dev.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-telcel/build-config/prod.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/client-telcel/build-config/qa.properties
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/libs/ui/res/values/configurations.xml

Removed unused webview.plan.management and webview_plan_management properties
------------------------------------------------------------------------
r1551569 | Vincent | 2015-11-27 09:38:09 +0000 (Fri, 27 Nov 2015) | 1 line
Changed paths:
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/settings.gradle

enable all clients
------------------------------------------------------------------------
r1551558 | Thomas | 2015-11-27 09:13:26 +0000 (Fri, 27 Nov 2015) | 1 line
Changed paths:
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/clients/common/scripts/clients.gradle

Chnaged jira url to htps
------------------------------------------------------------------------
r1551504 | /OU=Domain Control Validated/CN=svn.company.net | 2015-11-27 07:05:41 +0000 (Fri, 27 Nov 2015) | 1 line
Changed paths:
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/version.properties

[gradle-release] prepare for next development iteration
------------------------------------------------------------------------
r1551486 | Vincent | 2015-11-27 06:10:10 +0000 (Fri, 27 Nov 2015) | 1 line
Changed paths:
   M /cloud/personal/client-international/android/branches/android-15.2-solutions/settings.gradle

SFR-108 : preparing release for SFR
------------------------------------------------------------------------

test_CA4_bigdata.py
import unittest
from datetime import datetime
from CA4_bigdata import get_commits, read_file

print(datetime.strptime('2015-11-26', '%Y-%m-%d'))
class TestCommits(unittest.TestCase):

    def setUp(self):
        self.data = read_file('changes.txt')

    def test_number_of_lines(self):
        self.assertEqual(5255, len(self.data))

    def test_number_of_commits(self):
        commits = get_commits(self.data)
        self.assertEqual(422, len(commits))
        self.assertEqual('Thomas', commits[0]['author'])
        self.assertEqual('Jimmy', commits[420]['author'])
        self.assertEqual(['FTRPC-500: Frontier Android || Inconsistencey in My Activity screen',
				'Client used systemAttribute name="Creation-Date" instead of versionCreated as version created.'],
				commits[24]['comment'])
        self.assertEqual(['M /cloud/personal/client-international/android/branches/android-15.2-solutions/libs/model/src/com/biscay/client/android/model/util/sync/dv/SyncAdapter.java'],
                commits[20]['changed_path'])
    
    def test_statistics(self):    
        self.assertEqual(2,tot_lines.at['Dave','number_of_lines'])
        self.assertEqual(234,tot_lines.at['Thomas','number_of_lines'])
        self.assertEqual(80,tot_lines.at['Vincent','number_of_lines'])
        self.assertEqual(30,tot_date.at[datetime.strptime('2015-11-26', '%Y-%m-%d'),'number_of_lines'])
        self.assertEqual(20,tot_date.at[datetime.strptime('2015-11-20', '%Y-%m-%d'),'number_of_lines'])
        self.assertEqual(19,tot_date.at[datetime.strptime('2015-11-12', '%Y-%m-%d'),'number_of_lines'])
        self.assertEqual(5.552631578947368,release_no['number_of_lines'].mean())
        self.assertEqual(1.3151658767772512,df['number_of_lines'].mean())
        
if __name__ == '__main__':
    unittest.main()


CA4_bigdata.py

# coding: utf-8

# In[27]:


import pandas as pd
import numpy as np
import matplotlib.pyplot as plt


# In[28]:



def read_file(any_file):
    # use strip to strip out spaces and trim the line.
    return [line.strip() for line in open(any_file, 'r')]

def get_commits(data):
    sep = 72*'-'
    commits = []
    index = 0
    while index < len(data):
        try:
            # parse each of the commits and put them into a list of commits
            details = data[index + 1].split('|')
            # the author with spaces at end removed.
            commit = {'revision': details[0].strip(),
                'author': details[1].strip(),
                'date': details[2].strip().split(' ')[0],
                'time': details[2].strip().split(' ')[1],
                'number_of_lines': int(details[3].strip().split(' ')[0])
            }
            change_file_end_index = data.index('', index + 1)
            commit['changed_path'] = data[index + 3 : change_file_end_index]
            commit['comment'] = data[change_file_end_index + 1 : 
                    change_file_end_index + 1 + commit['number_of_lines']]
            # add details to the list of commits.
            commits.append(commit)
            index = data.index(sep, index + 1)
        except IndexError:
            index = len(data)
    return commits

def save_commits(commits, any_file):
    my_file = open(any_file, 'w')
    my_file.write("revision,author,date,time,number_of_lines,comment\n")
    for commit in commits:
        my_file.write(commit['revision'] + ',' + commit['author'] +
                ',' + commit['date'] + ',' + commit['time'] + ',' +
				str(commit['number_of_lines']) + ',' + ' '.join(commit['comment']) + '\n')
    my_file.close()

if __name__ == '__main__':
#    open the file - and read all of the lines.
    changes_file = 'changes.txt'
    data = read_file(changes_file)
    print (len(data))
    commits = get_commits(data)
    save_commits(commits, 'changes.csv')


# In[29]:


    df=pd.DataFrame(commits)


# In[30]:


    print(df.head())


# In[31]:


    print(df.columns)


# In[32]:


    df['date'] = pd.to_datetime(df['date'])
    df['time'] = pd.to_timedelta(df['time'])


# In[33]:


    name_lines=df.loc[:,['author','number_of_lines']]
    tot_lines = name_lines.groupby(['author']).sum()
    tot_lines.at['Dave','number_of_lines']


# In[34]:


    print(tot_lines)


# In[35]:


    tot_lines.plot(kind="bar")


# In[36]:


    #name_time=df.loc[:,['author','time']]
    #name_time.groupby(['author']).describe()




# In[37]:


    date_lines=df.loc[:,['date','number_of_lines']]
    tot_date=date_lines.groupby(['date']).sum()


# In[38]:


    print(tot_date.sort_values('number_of_lines', ascending=False).head())


# In[39]:


# Avarage number of realeases per day
    release_no=date_lines.groupby(['date']).count()
    print(release_no['number_of_lines'].mean())


# In[40]:


    print(df['number_of_lines'].mean())

